Sources/TuistGenerator/Mappers/ModuleMapMapper.swift

```diff
@@ -0,0 +1,123 @@
+/// It is required to avoid embedding the module map into the frameworks during cache operations, which would make the framework not portable, as
+/// the modulemap could contain absolute paths.
+public final class ModuleMapMapper: WorkspaceMapping {
+    private static let modulemapFileSetting = "MODULEMAP_FILE"
+    private static let otherSwiftFlagsSetting = "OTHER_SWIFT_FLAGS"
+
+    public init() {}
+
+    public func map(workspace: WorkspaceWithProjects) throws -> (WorkspaceWithProjects, [SideEffectDescriptor]) {
+        var targetToModuleMaps: [TargetID: Set<AbsolutePath>] = [:]
+        workspace.projects.forEach { project in
+            project.targets.forEach { target in
+                Self.dependenciesModuleMaps(workspace: workspace, project: project, target: target, targetToModuleMaps: &targetToModuleMaps)
+            }
+        }
+
+        var mappedWorkspace = workspace
+        for p in 0 ..< workspace.projects.count {
+            var mappedProject = workspace.projects[p]
+            for t in 0 ..< mappedProject.targets.count {
+                var mappedTarget = mappedProject.targets[t]
+                let targetID = TargetID(projectPath: mappedProject.path, targetName: mappedTarget.name)
+                var mappedSettingsDictionary = mappedTarget.settings?.base ?? [:]
+                guard mappedSettingsDictionary[Self.modulemapFileSetting] != nil || targetToModuleMaps[targetID] != nil else { continue }
+
+                if mappedSettingsDictionary[Self.modulemapFileSetting] != nil {
+                    mappedSettingsDictionary[Self.modulemapFileSetting] = nil
+                }
+
+                if let dependenciesModuleMaps = targetToModuleMaps[targetID] {
+                    var mappedSwiftCompilerFlags: [String]
+                    switch mappedSettingsDictionary[Self.otherSwiftFlagsSetting, default: .array(["$(inherited)"])] {
+                    case .array(let values):
+                        mappedSwiftCompilerFlags = values
+                    case .string(let value):
+                        mappedSwiftCompilerFlags = value.split(separator: " ").map(String.init)
+                    }
+
+                    for moduleMap in dependenciesModuleMaps.sorted() {
+                        mappedSwiftCompilerFlags.append(contentsOf: [
+                            "-Xcc",
+                            "-fmodule-map-file=$(SRCROOT)/\(moduleMap.relative(to: mappedProject.path))"
+                        ])
+                    }
+                    mappedSettingsDictionary[Self.otherSwiftFlagsSetting] = .array(mappedSwiftCompilerFlags)
+                }
+
+                mappedTarget.settings = (mappedTarget.settings ?? .default).with(base: mappedSettingsDictionary)
+                mappedProject.targets[t] = mappedTarget
+            }
+            mappedWorkspace.projects[p] = mappedProject
+        }
+        return (mappedWorkspace, [])
+    }
+
+    fileprivate static func dependenciesModuleMaps(
```

Comment:
```suggestion
    private static func dependenciesModuleMaps(
```

Reply:
can you add comment to this function as a high-level overview about what it does and why?